woocommerce/woocommerce-ios
# [Mobile Payments] Implement support for emailing a receipt right after a payment has been captured

Closes #4014 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️
⚠️ Testing this PR requires a hardware reader and uploading a pre-release version of WCPay to a test site⚠️


https://user-images.githubusercontent.com/2722505/117327291-6df81380-ae60-11eb-9793-70405cd0a8fc.mov

## Changes
* Make ReceiptRenderer public. Not ideal, but, in my opinion, not that terrible either. That being said, we could maybe declare two different "receipt renderers", one for printed receipts and another one for emailed receipts, and extract the code that generates the body of the receipt as HTML to an aggregated entity. Let's see how it goes.
* Move the "footer" (which was pretty much just the two fields we need to include for regulatory reasons) to the body of the receipt.
* Add support for a new Action to the ReceiptsStore to generate the body of the receipt as HTML
* Updates to OrderDetailsViewController and OrderDetailsViewModel to trigger that action, add localized strings, present the mail controller... All the good stuff.


## How to test
Testing requires:

1. Setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. The pre-release version of WCPay that implements the endpoint that captures the payment id (see p91TBi-54R#comment-4441-p2 for a downloadable zip)
3. An external hardware reader.

* Checkout the branch, run bundle exec pod install, just for kicks and giggles.
* Build and run on a device.
* Switch an external card reader on. The only card reader supported by now is the BBPOS Chipper 2X BT.
* Log in to an account that matches a store that has been configured for testing.
* Navigate to Settings in the app. (Tap the gear button in the top right)
* In the "Store Settings" section, tap "Manage Card Reader".
* Make sure the card reader is on and the blue light is blinking (we don't do any error handling yet, so it's important that things are setup for the happy path)
* Tap "Connect card reader"
* Wait until the app suggests one reader to connect to.
* Tap "Connect to reader"
* Make sure the reader light turns solid blue.
* Navigate to an order eligible for Card Present Payments (it has to be a store enrolled to WCPay, and the order needs to have as payment method associated Cash on Delivery or none).
* Tap the "Collect Payment" button.
* Tap the card to the reader when requested.
* If everything goes well, the card should be charged, and the payment should be completed.
* Tap the "Email Receipt" button
* The standard iOS email composing sheet is presented. If there is a customer email address on file, it should be pre-populated with it. The subject should read "Your receipt from STORE_NAME". The receipt should contain, at this point, a list of the order items, the total, and two fields we need to include into the receipt for regulatory reasons.


Update release notes:
